src/libs/ajax/compute-image-providers/ComputeImageProvider.ts

```diff
@@ -0,0 +1,93 @@
+import { ComputeImage } from 'src/analysis/useComputeImages';
+import {
+  getToolLabelForImage,
+  isRuntimeToolLabel,
+  runtimeToolLabels,
+  terraSupportedRuntimeImageIds,
+} from 'src/analysis/utils/tool-utils';
+import { Ajax } from 'src/libs/ajax';
+import { getConfig } from 'src/libs/config';
+
+export interface ComputeImageProviderContract {
+  listImages: (googleProject: string, signal?: AbortSignal) => Promise<ComputeImage[]>;
+}
+
+/**
+ * Refers to a docker image of a Terra VM.
+ * Sample response:
+     {
+        "id": "terra-jupyter-bioconductor",
+        "image": "us.gcr.io/broad-dsp-gcr-public/terra-jupyter-bioconductor:2.2.1",
+        "label": "R / Bioconductor: (Python 3.10.11, R 4.3.1, Bioconductor 3.17, tidyverse 2.0.0)",
+        "packages": "https://storage.googleapis.com/terra-docker-image-documentation/terra-jupyter-bioconductor-2.2.1-versions.json",
+        "requiresSpark": false,
+        "updated": "2023-07-13",
+        "version": "2.2.1"
+    },
+ */
+export interface ComputeImageRaw {
+  id: string;
+  image: string;
+  isCommunity?: boolean;
+  isRStudio?: boolean;
+  label: string;
+  packages: string;
+  requiresSpark: boolean;
+  updated: string;
+  version: string;
+}
+
+const normalizeImage: (rawImage: ComputeImageRaw) => ComputeImage = (rawImage) => {
+  const toolLabel = getToolLabelForImage(rawImage.id);
+  if (toolLabel && !isRuntimeToolLabel(toolLabel)) {
+    throw Error('Compute images are not supported for non-Runtime tools');
+  }
+
+  /* eslint-disable */
+  //prettier-ignore
```

Comment:
```suggestion
```

Are these needed? I tried removing them and it looks like nothing changed. If the linter does need to be disabled, better to disable it for a specific line or disable a specific rule vs the blanket disable for the rest of the file.

Reply:
Oh I was trying to disable it only for this line! Otherwise the linter puts this destructuring assignment on one line which I find dangerously unreadable for this normalization function

Reply:
How would I disable just this line's linting?

Reply:
```
// eslint-disable-next-line prettier/prettier
```
https://eslint.org/docs/latest/use/configure/rules#using-configuration-comments-1

However, I find it's often a losing battle to attempt to override the formatter. This would disable the ESLint rule and it's auto fix, but running Prettier outside of ESLint would still format this.

I'm surprised though. I'm not seeing this get formatted. And Prettier _shouldn't_ be changing this to one line:
> Prettier instead keeps objects multiline if there’s a newline between the { and the first key in the original source code.
https://prettier.io/docs/en/rationale.html#multi-line-objects

Reply:
weird! It might also be that this was only formatting to one line before I added the latest var, isTerraSupported. What do you use to lint? Run `yarn lint`?

Reply:
`yarn lint`, but mainly the ESLint plugin for VS Code.